This patch modifies the SIP return message in two ways:

* Field CY: Now contains borrowernumber, instead of cardnumber
* Field AY: Biblionumber of item

These mods are needed for hold-transfer-slip.pl to print a slip

diff --git a/C4/SIP/ILS/Item.pm b/C4/SIP/ILS/Item.pm
index 578f382..d797b58 100644
--- a/C4/SIP/ILS/Item.pm
+++ b/C4/SIP/ILS/Item.pm
@@ -185,13 +185,17 @@ sub hold_patron_name {
 sub hold_patron_bcode {
     my $self = shift;
     my $borrowernumber = (@_ ? shift: $self->hold_patron_id()) or return;
-    my $holder = GetMember(borrowernumber => $borrowernumber);
-    if ($holder) {
-        if ($holder->{cardnumber}) {
-            return $holder->{cardnumber};
-        }
-    }
-    return;
+    # Deichman modification:
+    # We need borrowernumber to generate a hold-transfer-slip,
+    # so we return borrowernumber instead of cardnumber
+
+    #my $holder = GetMember(borrowernumber => $borrowernumber);
+    #if ($holder) {
+    #    if ($holder->{cardnumber}) {
+    #        return $holder->{cardnumber};
+    #    }
+    #}
+    return $borrowernumber;
 }

 sub destination_loc {
diff --git a/C4/SIP/Sip/MsgType.pm b/C4/SIP/Sip/MsgType.pm
index 4e406e2..fce1299 100644
--- a/C4/SIP/Sip/MsgType.pm
+++ b/C4/SIP/Sip/MsgType.pm
@@ -680,6 +680,7 @@ sub handle_checkin {
             $resp .= maybe_add( FID_DESTINATION_LOCATION, $item->destination_loc );
             $resp .= maybe_add( FID_HOLD_PATRON_ID,       $item->hold_patron_bcode );
             $resp .= maybe_add( FID_HOLD_PATRON_NAME,     $item->hold_patron_name );
+            $resp .= maybe_add( FID_SEQNO,                $item->biblionumber ); # Deicman modification:  We need biblionumber to generate hold-transfer-slip
             if ( $status->hold and $status->hold->{branchcode} ne $item->destination_loc ) {
                 warn 'SIP hold mismatch: $status->hold->{branchcode}=' . $status->hold->{branchcode} . '; $item->destination_loc=' . $item->destination_loc;
